src/back/mod.rs

```diff
@@ -1,5 +1,10 @@
 //! Functions which export shader modules into binary and text formats.
 
+#[cfg(feature = "deserialize")]
```

Comment:
nit: if it's just one structure, much easier to write it as `serde::Deserialize` there direction instead of adding these lines